config/v1/types_infrastructure.go

```diff
@@ -500,6 +508,36 @@ type AWSResourceTag struct {
 	Value string `json:"value"`
 }
 
+// DNSConfigurationType contains information about who configures DNS for the
+// cluster.
+// +union
+type DNSConfigurationType struct {
+	// provider determines which DNS solution is in use for this cluster.
+	// When the user wants to use their own DNS solution, the `provider` is set to
+	// "UserProvided".
+	// When the cluster configures the cloud DNS, this field is set to
+	// "CloudProvided". The default is "CloudProvided".
+	// +default="CloudProvided"
+	// +kubebuilder:default:="CloudProvided"
+	// +kubebuilder:validation:Enum="UserProvided";"CloudProvided"
```

Comment:
After discussion with @patrickdillon , we came to the conclusion that "UserProvided" does not give an accurate complete picture. So, proposal is to update this string to "UserAndInClusterProvided". 
@JoelSpeed so I am currently working on updating the PR.

Reply:
The 2 Provider values are "Default" and "UserAndClusterProvided" to represent the default and hybrid (where user provides DNS for API and API-Int and the cluster has to provide DNS for *.apps during cluster bring up) cases  respectively.